pkg/config/applications/application.go

```
@@ -0,0 +1,550 @@
+	TotalCount int      `json:"totalCount" yaml:"totalCount"`
+}
+
+type Embedded struct {
+	Applications []*Application `json:"applications" yaml:"applications"`
+}
+
+type Application struct {
+	Name                   string                 `json:"name" yaml:"name"`
+	TemplateID             string                 `json:"templateId" yaml:"templateId"`
+	Links                  Links                  `json:"_links" yaml:"_links"`
+	Providers              Providers              `json:"providers" yaml:"providers"`
+	Provisioning           Provisioning           `json:"provisioning" yaml:"provisioning"`
+	AttributeMappings      []*AttributeMapping    `json:"attributeMappings" yaml:"attributeMappings,omitempty"`
+	ApplicationState       bool                   `json:"applicationState" yaml:"applicationState,omitempty"`
+	ApprovalRequired       bool                   `json:"approvalRequired" yaml:"approvalRequired,omitempty"`
+	SignonState            bool                   `json:"signonState" yaml:"signonState,omitempty"`
+	Description            string                 `json:"description" yaml:"description,omitempty"`
+	ProvisioningMode       string                 `json:"provisioningMode" yaml:"provisioningMode,omitempty"`
+	IdentitySources        []string               `json:"identitySources" yaml:"identitySources,omitempty"`
+	VisibleOnLaunchpad     bool                   `json:"visibleOnLaunchpad" yaml:"visibleOnLaunchpad,omitempty"`
+	Customization          Customization          `json:"customization" yaml:"customization,omitempty"`
+	DevportalSettings      DevportalSettings      `json:"devportalSettings" yaml:"devportalSettings,omitempty"`
+	APIAccessClients       []*APIAccessClients    `json:"apiAccessClients" yaml:"apiAccessClients,omitempty"`
+	CustomIcon             string                 `json:"customIcon" yaml:"customIcon,omitempty"`
+	DefaultIcon            string                 `json:"defaultIcon" yaml:"defaultIcon,omitempty"`
```

Is `defaultIcon` useful? If not, let's hide it away. I believe you can't PUT this back.

---

It is kept in order to receive the response, as we can't PUT it back, we can remove that thing in the boilerplate generation.

---

The expected usage of this capability is to do a Get then store the yaml file. Then modify it and Put. The code needs to handle removal of things that the API won't accept.

---

Okay I will remove it